### PR TITLE
Add latest k8s versions in testing

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?

This PR adds more k8s versions for testing.

### CODE REVIEW

**Review by MichaelKatsoulis — APPROVED** [review]

**Comment by ChrsMark** [comment]
It seems that default ports for proxy and scheduler have been changed in latest versions. We need to tune them accordingly for our test.

**Comment by ChrsMark** [comment]
I wonder why the config for `proxy` metricset targets `10252` port at https://github.com/elastic/beats/blob/64f6e809ed8427741a9adfaa4bcde7c39245f02a/metricbeat/module/kubernetes/test/integration.go#L74, while the port should be `10249` like at https://github.com/elastic/beats/blob/2a0e0998243db56d0bfd681ff3cb06183166bdd5/metricbeat/module/kubernetes/_meta/config.reference.yml#L76.

Also we need to find a way to set the proper ports for newest versions of k8s for `scheduler` and `controllermanger` since pure http is not support anymore and we should target secure ports now (see https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/).

@jsoriano do you have any ideas on how we could check for what version we are running and set the config accordingly at https://github.com/elastic/beats/blob/64f6e809ed8427741a9adfaa4bcde7c39245f02a/metricbeat/module/kubernetes/test/integration.go#L79?

File: deploy/kubernetes/Jenkinsfile.yml [hunk 1]
```diff
@@ -18,5 +18,5 @@ stages:
             make check-no-changes;
         stage: lint
     k8sTest:
-        k8sTest: "v1.18.2,v1.17.2,v1.16.4,v1.15.7,v1.14.10"
+        k8sTest: "v1.20.7,v1.19.11,v1.18.2,v1.17.2,v1.16.4,v1.15.7,v1.14.10"
```
Comment (jsoriano) [inline deploy/kubernetes/Jenkinsfile.yml:21]: It'd be nice to support 1.21 too :slightly_smiling_face:

Comment (jsoriano) [inline deploy/kubernetes/Jenkinsfile.yml:21]: Maybe we can start dropping some of the old versions from the matrix? I would leave the older one we support, but we can consider removing some of the others.

**Review by jsoriano — COMMENTED** [review]
> @jsoriano do you have any ideas on how we could check for what version we are running and set the config accordingly?

More than checking for the version, we might explicitly provide it via environment variables or something, the job invoking the test knows the version, so it could set an environment variable, and then in the methods to get the config we can chose what config to use depending on the version.

Other option is to configure the services to listen always in the same ports, it is possible to [customize the control plane with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/), so invoking kind with something like this config might do the trick for the scheduler (not tested):
```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: ClusterConfiguration
    scheduler:
      extraArgs:
        secure-port: "0"
        port: "10251"
```
Or we can try to enable always the new port, I guess that by setting the `secure-port` flag. But maybe this doesn't work if older versions don't have this flag.

(Take into account that it is not only a port change, but also a change from HTTP to HTTPS.)

**Comment by jsoriano** [comment]
> @jsoriano do you have any ideas on how we could check for what version we are running and set the config accordingly?

I have been trying some things and we cannot use only the old port because the scheduler pod has a healthcheck for the new one.

Good thing about this is that, if needed, we can use the token in `/var/run/secrets/kubernetes.io/serviceaccount/token` to query the API and get the port from the health check.

This config works to start with both ports, also in the oldest version:
```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: ClusterConfiguration
    scheduler:
      extraArgs:
        bind-address: "0.0.0.0"
        port: "10251"
        secure-port: "10259"
```

**Comment by ChrsMark** [comment]
👍🏼 @jsoriano thanks! In integrations we access the endpoint using the token too -> https://github.com/elastic/integrations/blob/master/packages/kubernetes/data_stream/scheduler/_dev/test/system/test-default-config.yml#L7.

However there we test only with latest version. I think the patch for kind will do the trick!

File: metricbeat/module/kubernetes/controllermanager/controllermanager_integration_test.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ import (
 )
 
 func TestFetchMetricset(t *testing.T) {
-	config := test.GetAPIServerConfig(t, "controllermanager")
+	config := test.GetControllerManagerConfig(t, "controllermanager")
```
Comment (MichaelKatsoulis) [inline metricbeat/module/kubernetes/controllermanager/controllermanager_integration_test.go:32]: 👍

File: deploy/kubernetes/Jenkinsfile.yml (hunk 1 above)
Comment (ChrsMark) [reply deploy/kubernetes/Jenkinsfile.yml]: I removed 1.15 and 1.16

Comment (ChrsMark) [reply deploy/kubernetes/Jenkinsfile.yml]: kind version required an update to work with 1.21

**Review by kuisathaverat — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]
LGTM if it looks good to jenkins :slightly_smiling_face:

**Comment by ChrsMark** [comment]
For reference: Versions of k8s should be specified as per https://github.com/kubernetes-sigs/kind/releases/tag/v0.11.1

**Pull request merged**